Tests/FluentAssertions.Specs/Numeric/ComparableSpecs.cs

```diff
@@ -103,6 +103,67 @@ public void When_two_unequal_objects_should_not_be_equal_it_should_not_throw()
         }
     }
 
+    public class BeOneOf
+    {
+        [Fact]
+        public void When_a_value_is_not_one_of_the_specified_values_it_should_throw()
+        {
+            // Arrange
+            var value = new ComparableOfInt(3);
+
+            // Act
+            Action act = () => value.Should().BeOneOf(new ComparableOfInt(4), new ComparableOfInt(5));
+
+            // Assert
+            act
+                .Should().Throw<XunitException>()
+                .WithMessage("Expected value to be one of {4, 5}, but found 3.");
+        }
+
+        [Fact]
+        public void When_a_value_is_not_one_of_the_specified_values_it_should_throw_with_descriptive_message()
+        {
+            // Arrange
+            var value = new ComparableOfInt(3);
+
+            // Act
+            Action act = () => value.Should().BeOneOf(new[] { new ComparableOfInt(4), new ComparableOfInt(5) }, "because those are the valid values");
+
+            // Assert
+            act
+                .Should().Throw<XunitException>()
+                .WithMessage("Expected value to be one of {4, 5} because those are the valid values, but found 3.");
+        }
+
+        [Fact]
+        public void When_a_value_is_comparable_to_but_a_different_instance_of_one_of_the_specified_values_it_should_fail()
+        {
+            // Arrange
+            var value = new ComparableOfInt(4);
+
+            // Act
+            Action act = () => value.Should().BeOneOf(new ComparableOfInt(4), new ComparableOfInt(5));
+
+            // Assert
+            act
+                .Should().Throw<XunitException>()
+                .WithMessage("Expected value to be one of {4, 5}, but found 4.");
+        }
+
+        [Fact]
+        public void When_a_value_is_the_same_instance_as_one_of_the_specified_values_it_should_succeed()
```

Comment:
I was wondering what this test was doing.
It seems we have a hole in our test coverage as `EquatableOfInt.Should()` is picked up by `ObjectAssertions` and not `ComparableTypeAssertions` as it does not implement `IComparable<T>`.
So the tests in `ComparableSpecs` of `[Not]Be` actually tests `ObjectAssertions.[Not]Be`.

I suggest making these changes to `EquatableOfInt` and use it instead of `ComparableInt` for all the added tests.
```diff
- public class EquatableOfInt
+ public class EquatableOfInt : IComparable<EquatableOfInt>
  {
+     int IComparable<EquatableOfInt>.CompareTo(EquatableOfInt other) =>
+        throw new NotSupportedException("This type is meant for assertions using Equals()");
  }
```

Reply:
OK, would it not be better for you to make these changes directly after mine as you understand this bit of the code base better than I do?

Reply:
Then I'll rather make the changes before you, so you're PR is correct on merge.
I'll make a PR for the fix of `EquatableOfInt`. 
It might take a little time before it is merged as @dennisdoomen is busy speaking at conferences.

Reply:
#2036

Reply:
#2036 has been merged, so you can rebase on `develop` and change these tests to use `EquatableOfInt` instead and not reuse `value` in this test.